src/main/scala/za/co/absa/atum/web/api/controller/FlowController.scala

```diff
@@ -17,12 +17,118 @@
 package za.co.absa.atum.web.api.controller
 
 import org.springframework.beans.factory.annotation.Autowired
-import org.springframework.web.bind.annotation.{RequestMapping, RestController}
+import org.springframework.http.{HttpStatus, ResponseEntity}
+import org.springframework.web.bind.annotation._
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder
+import za.co.absa.atum.web.api.controller.BaseApiController.{LimitParamName, OffsetParamName}
+import za.co.absa.atum.web.api.controller.FlowController.FlowDefParamName
+import za.co.absa.atum.web.api.implicits._
+import za.co.absa.atum.web.api.payload.MessagePayload
 import za.co.absa.atum.web.api.service.FlowService
+import za.co.absa.atum.web.model._
+
+import java.net.URI
+import java.util.UUID
+import java.util.concurrent.CompletableFuture
+import javax.servlet.http.HttpServletRequest
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.collection.JavaConverters._
 
 @RestController
 @RequestMapping(Array("/api/flows"))
-class FlowController @Autowired()(flowService: FlowService)
-  extends BaseApiController(flowService) {
+class FlowController @Autowired()(flowService: FlowService) extends BaseApiController(flowService) {
+
+  @PutMapping(Array("/{id}"))
+  @ResponseStatus(HttpStatus.OK)
+  def update(@PathVariable id: UUID, @RequestBody flow: Flow): CompletableFuture[MessagePayload] = {
+    require(flow.id.isEmpty, s"A $entityName update payload must have no id (it is given in the URL)!")
```

Comment:
I tried to look up some "best practices" on this, but I could not find anything conclusive. It would seem that the consensus is a bit towards not sending back the updated entity or even "204 No content".

In cases the caller needs the entity, they can get it via GET afterwards. If they don't, they don't need to process, a potentially large, payload.

Let's see what we end up discussing today.